jupyterlab_gitsync/src/service/git.ts

```diff
@@ -0,0 +1,75 @@
+import { requestAPI } from './request_api';
+import { ISignal, Signal } from '@lumino/signaling';
+
+/**
+ *
+ * Class responsible for requesting git handlers on the server side
+ * to synchronize local repository with remote repository
+ *
+ */
+
+export class GitManager {
+  // Member Fields
+  private _path: string;
+  private _executablePath: string;
+  private _fileConflicts: string[] = undefined;
+  private _syncCompleted: Signal<this, void> = new Signal<this, void>(this);
+  private _mergeConflict: Signal<this, void> = new Signal<this, void>(this);
+
+  constructor(path: string) {
+    this._setup(path);
+  }
+
+  get path(): string {
+    return this._path;
+  }
+
+  get fileConflicts(): string[] {
+    return this._fileConflicts;
+  }
+
+  get syncCompleted(): ISignal<this, void> {
+    return this._syncCompleted;
+  }
+
+  get mergeConflict(): ISignal<this, void> {
+    return this._mergeConflict;
+  }
+
+  async sync(args: string = null, emit = true) {
+    const init: RequestInit = {
+      method: 'POST',
+      body: JSON.stringify({
+        path: this._path,
+        ex_path: this._executablePath,
+        options: ['origin', 'ashleyswang/master'],
```

Comment:
please don't hardcode values. let's find a way to pass in, or add #TODO. It will be hard to debug.

Reply:
Changed to be passed in as a parameter to class. Hardcoded values now on `index.ts` with a TO DO for adding configuration options.